src/Avalonia.Controls/WindowIcon.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Avalonia.Platform;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Avalonia.Media.Imaging;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// Represents an icon for a window.
+    /// </summary>
+    public class WindowIcon
```

Comment:
The special thing about icons is that they can have multiple sizes in a single file, correct? I believe this is also something that taskbar/desktop icons etc have? Bearing this in mind, would `ApplicationIcon` not make more sense than `WindowIcon`?